asset-transfer-basic/application-typescript/src/app.ts

```diff
@@ -3,28 +3,66 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-import { Gateway, GatewayOptions } from 'fabric-network';
+
+import * as grpc from '@grpc/grpc-js';
+import { ServiceClient } from '@grpc/grpc-js/build/src/make-client';
+import * as crypto from 'crypto';
+import { connect, Identity, Signer, signers } from 'fabric-gateway';
+import { promises as fs } from 'fs';
 import * as path from 'path';
-import { buildCCPOrg1, buildWallet, prettyJSONString } from './utils//AppUtil';
-import { buildCAClient, enrollAdmin, registerAndEnrollUser } from './utils/CAUtil';
 
 const channelName = 'mychannel';
 const chaincodeName = 'basic';
-const mspOrg1 = 'Org1MSP';
-const walletPath = path.join(__dirname, 'wallet');
-const org1UserId = 'appUser';
+const mspId = 'Org1MSP';
+
+const cryptoPath = path.resolve(
+    __dirname,
+    '..',
+    '..',
+    '..',
+    'test-network',
+    'organizations',
+    'peerOrganizations',
+    'org1.example.com',
+);
+
+const keyPath = path.resolve(
+    cryptoPath,
+    'users',
+    'Admin@org1.example.com',
+    'msp',
+    'keystore',
+    'priv_sk',
+
+);
+
+const certPath = path.resolve(
+    cryptoPath,
+    'users',
+    'Admin@org1.example.com',
+    'msp',
+    'signcerts',
+    'Admin@org1.example.com-cert.pem',
+);
+const tlsCertPath = path.resolve(
+    cryptoPath,
+    'peers',
+    'peer0.org1.example.com',
+    'tls',
+    'ca.crt',
+);
+const peerEndpoint = 'localhost:7051';
 
 // pre-requisites:
-// - fabric-sample two organization test-network setup with two peers, ordering service,
-//   and 2 certificate authorities
+// - fabric-sample two organization test-network setup with two peers and ordering service
 //         ===> from directory /fabric-samples/test-network
-//         ./network.sh up createChannel -ca
```

Comment:
Why remove the CA option and related register/enroll utilities?
Won't we still want to demonstrate how to use fabric-ca-client to interact with CAs?

Reply:
I agree that demonstrating interaction with the CA is useful. I'm not sure if the mechanism we should be promoting is the SDKs or if it would be better to promote the use of the CLI. The Fabric Gateway API that this PR is demonstrating does not have any API for interacting with the CA.

One thing I really dislike about the current samples is how the user identity is enrolled with the CA as part of the same code (and often the same function) as the transactional logic. This is absolutely not how real applications should be written and actively promotes bad practices. Countless times I have seen application code that does enrolment for client identities every time the application is run. In other words, create new client credentials for every invocation of the application. Identities should be created as a separate administrative step (and probably by an administrator) prior running client applications that make use of those client identities.

Reply:
ok, i'm good with removing CA register/enrollment from most of the samples, as long as we have at least one sample that focuses on register/enroll with fabric-ca-client.